File: test/ContactFormTest.java
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import static junit.framework.TestCase.assertEquals;
import static org.junit.Assert.assertTrue;

public class ContactFormTest {
    static WebDriver driver;

    @BeforeClass
    static public void beforeClass() {
        driver = new FirefoxDriver();
        driver.get("https://www.sweetlightstudios.com/contact/");
        String titleText = driver.getTitle();
        System.out.println("titleText: " + titleText);
    }

    @Test
    public void contactFormButtonTest(){
        fillOutForm();
        driver.findElement(By.cssSelector(".contact-button")).click();
        new WebDriverWait(driver,10).until(ExpectedConditions.titleIs("Formspree"));
        driver.navigate().back(); // come back to Sweet Light Studios
        new WebDriverWait(driver,10).until(ExpectedConditions.titleContains("South San Francisco"));
        driver.findElement(By.cssSelector("article.contact")).isDisplayed(); // verify contact article is displayed
    }

    @Test  // This isn't really needed as it is testing the browser ability and not the form
    public void contactFormSpaceBarTest(){ // test if can submit via space bar for compliance
        fillOutForm();
        driver.findElement(By.cssSelector(".contact-button")).sendKeys(Keys.SPACE);
        new WebDriverWait(driver,10).until(ExpectedConditions.titleIs("Formspree"));
        driver.navigate().back(); // come back to Sweet Light Studios
        new WebDriverWait(driver,10).until(ExpectedConditions.titleContains("South San Francisco"));
        driver.findElement(By.cssSelector("article.contact")).isDisplayed(); // verify contact article is displayed
    }

    @Test
    public void contactFormSubmitTest(){
        fillOutForm();
        driver.findElement(By.cssSelector("form[action=\"https://formspree.io/rhbourbonnais@yahoo.com\"]")).submit();
        new WebDriverWait(driver,10).until(ExpectedConditions.titleIs("Formspree"));
        driver.navigate().back(); // come back to Sweet Light Studios
        new WebDriverWait(driver,10).until(ExpectedConditions.titleContains("South San Francisco"));
        driver.findElement(By.cssSelector("article.contact")).isDisplayed(); // verify contact article is displayed
    }

    private void fillOutForm() {
        String testEmail = "testytest@gmail.com";
        String testName = "Mr. Testy Test";
        String radioButtonSelector = "input[value=\"Personal Branding\"]";
        String testPhone = "831-555-1212";
        String testInfo = "I want a new picture for FB";
        driver.findElement(By.cssSelector("article.contact")).isDisplayed(); // contact article

        // fill out form
        driver.findElement(By.id("formControlsEmail")).sendKeys(testEmail);
        driver.findElement(By.name("name")).sendKeys(testName);
        driver.findElement(By.cssSelector(radioButtonSelector)).click();
        driver.findElement(By.name("phone")).sendKeys(testPhone);
        driver.findElement(By.name("additionalInfo")).sendKeys(testInfo);

        // verify form filled out
        assertEquals("Email Not Correct", testEmail, driver.findElement(By.id("formControlsEmail")).getAttribute("value"));
        assertEquals("Name Not Correct", testName, driver.findElement(By.name("name")).getAttribute("value"));
        assertTrue("Radio Button not selected", driver.findElement(By.cssSelector(radioButtonSelector)).isSelected() );
        assertEquals("Phone Not Correct", testPhone, driver.findElement(By.name("phone")).getAttribute("value"));
        assertEquals("Info Not Correct", testInfo, driver.findElement(By.name("additionalInfo")).getAttribute("value"));
    }

    @AfterClass
    static public void afterClass() {
        driver.quit();
    }
}

File: test/ContactForm-test.js
// Run from command line in the test folder via
// $ mocha ContactForm-test.js --timeout=20000
// see ContactFormTest.java for Rosetta stone

const webdriver = require('selenium-webdriver');
const assert = require('assert');

const By = webdriver.By;       // DRY
const until = webdriver.until;

const driver = new webdriver.Builder()
    .forBrowser('chrome')
    .build();

describe('Contact Form to Formspree', () => {

  process.on('unhandledRejection', error => {
    throw(error); // promote promise warning to mocha error
  });
  
  beforeEach(async function() {
    await driver.navigate().to('https://www.sweetlightstudios.com/contact');
    await driver.wait(until.titleContains('South San Francisco'),10000);
  });

  it('Contact Form Button Test', async function() {
    await fillOutForm();
    await driver.findElement(By.css(".contact-button")).click();
    await driver.wait(until.titleIs('Formspree'),10000);
    await driver.navigate().back(); // come back to Sweet Light Studios
    await driver.wait(until.titleContains('South San Francisco'),10000);
    await driver.wait(until.elementIsVisible(await driver.findElement(By.css("article.contact"))),10000);
  }); 

  it('Contact Form Submit Test', async function() {
    await fillOutForm();
    //await driver.findElement(By.css(".contact-button")).click();
    await driver.findElement(By.css("form[action=\"https://formspree.io/rhbourbonnais@yahoo.com\"]")).submit();
   
    await driver.wait(until.titleIs('Formspree'),10000);
    await driver.navigate().back(); // come back to Sweet Light Studios
    await driver.wait(until.titleContains('South San Francisco'),10000);
    await driver.wait(until.elementIsVisible(await driver.findElement(By.css("article.contact"))),10000);
  }); 

  const fillOutForm = (async function() {
    const testEmail = "testytest@gmail.com";
    const testName = "Mr. Testy Test";
    const radioButtonSelector = "input[value=\"Personal Branding\"]";
    const testPhone = "831-555-1212";
    const testInfo = "I want a new picture for FB";
    await driver.wait(until.elementIsVisible(await driver.findElement(By.css("article.contact"))),10000);

    // fill out form
    await driver.findElement(By.id("formControlsEmail")).sendKeys(testEmail);
    await driver.findElement(By.name("name")).sendKeys(testName);
    await driver.findElement(By.css(radioButtonSelector)).click();
    await driver.findElement(By.name("phone")).sendKeys(testPhone);
    await driver.findElement(By.name("additionalInfo")).sendKeys(testInfo);

    // verify form filled out
    assert.equal(
      await driver.findElement(By.id("formControlsEmail")).getAttribute("value"), 
      testEmail, 
      "Email Not Correct"
    );

    assert.equal( 
      await driver.findElement(By.name("name")).getAttribute("value"),
      testName,
      "Name Not Correct"
    );

    assert(
      await driver.findElement(By.css(radioButtonSelector)).isSelected(),
      "Radio Button not selected" 
    );

    assert.equal(
      await driver.findElement(By.name("phone")).getAttribute("value"),
      testPhone, 
      "Phone Not Correct"
    );

    assert.equal(
      await driver.findElement(By.name("additionalInfo")).getAttribute("value"),
      testInfo, 
      "Info Not Correct"
    );

  });

  after(async function() {
    await driver.quit();
  });

});
